app/src/main/java/com/zeapo/pwdstore/PasswordStore.kt

```diff
@@ -591,6 +592,32 @@ class PasswordStore : AppCompatActivity(R.layout.activity_pwdstore) {
         startActivityForResult(intent, REQUEST_CODE_SELECT_FOLDER)
     }
 
+    fun renameCategory(values: Stack<PasswordItem>) {
+        if (values.isNotEmpty()) {
+            val value = values.pop()
+            if (value.type == PasswordItem.TYPE_CATEGORY) {
+                val view = layoutInflater.inflate(R.layout.folder_dialog_fragment, null)
+                val newCategoryEditText = view.findViewById<TextInputEditText>(R.id.folder_name_text)
+
+                MaterialAlertDialogBuilder(this)
+                    .setTitle(R.string.title_rename_folder)
+                    .setView(view)
+                    .setMessage(getString(R.string.message_rename_folder, value.name))
+                    .setPositiveButton(R.string.dialog_ok) { _, _ ->
+                        val newCategory = File("${value.file.parent}/${newCategoryEditText.text}")
+                        value.file.renameTo(newCategory)
+                        commitChange(resources.getString(R.string.git_commit_move_text, value.name, newCategory.name))
+                        renameCategory(values)
+                    }
+                    .setNegativeButton(R.string.dialog_cancel) { _, _ ->
+                        renameCategory(values)
+                    }
+                    .show()
+            }
+        } else
+            refreshPasswordList()
```

Comment:
```suggestion
        } else {
            refreshPasswordList()
        }
```

I just prefer it this way